repo-impl/src/main/java/com/gtxtreme/template/repo/ContentRepoImpl.kt

```diff
@@ -0,0 +1,19 @@
+package com.gtxtreme.template.repo
+
+import com.gtxtreme.template.content.DomainContentMapper
+import com.gtxtreme.template.domain.content.Content
+import com.gtxtreme.template.service.itunecontent.MediaContentRemoteService
+
+class ContentRepoImpl(
+    private val contentRemoteService: MediaContentRemoteService,
+    private val domainContentMapper: DomainContentMapper
+) :
+    ContentRepository {
+    override suspend fun getAllContentByArtistName(artistName: String): List<Content> =
+
+        if (contentRemoteService.getMediaDetailByAuthor(artistName).resultCount == 0) {
+            emptyList()
+        } else {
+            domainContentMapper.map(contentRemoteService.getMediaDetailByAuthor(artistName).results)
+        }
```

Comment:
We should extract the api call to a separate line. In this case if the resultCount is not 0 the api call will happen twice.
```suggestion
       val response = contentRemoteService.getMediaDetailByAuthor(artistName)
        if (response.resultCount == 0) {
            emptyList()
        } else {
            domainContentMapper.map(response.results)
        }
```